platforms/core-runtime/build-configuration/src/main/java/org/gradle/buildconfiguration/tasks/UpdateDaemonJvm.java

```diff
@@ -56,39 +63,63 @@
 @Incubating
 public abstract class UpdateDaemonJvm extends DefaultTask {
 
+    /**
+     * The problem id for task configuration problems.
+     *
+     * @since 8.13
+     */
+    public static final ProblemId TASK_CONFIGURATION_PROBLEM_ID = ProblemId.create("task-configuration", "Invalid task configuration", GradleCoreProblemGroup.daemonToolchain().configurationGeneration());
+
     private final DaemonJvmPropertiesModifier daemonJvmPropertiesModifier;
+    private final Property<String> jvmVendorDeprecated;
+    private final ProblemReporter problemsReporter;
 
     /**
      * Constructor.
      *
      * @since 8.8
      */
     @Inject
-    public UpdateDaemonJvm(DaemonJvmPropertiesModifier daemonJvmPropertiesModifier) {
+    public UpdateDaemonJvm(DaemonJvmPropertiesModifier daemonJvmPropertiesModifier, PropertyFactory propertyFactory, Problems problems) {
         this.daemonJvmPropertiesModifier = daemonJvmPropertiesModifier;
+        jvmVendorDeprecated = propertyFactory.property(String.class);
+        problemsReporter = problems.getReporter();
     }
 
     @TaskAction
     void generate() {
         IncubationLogger.incubatingFeatureUsed("Daemon JVM criteria");
 
-        final String jvmVendor;
-        Provider<JvmVendorSpec> vendorSpec = getJvmVendor().map(JvmVendorSpec::of);
-        if (vendorSpec.isPresent()) {
-            // TODO change this to something else, we should serialize the spec, not just the vendor string
-            jvmVendor = getJvmVendor().get();
+        handleDeprecatedJvmVendor();
+
+        final String jvmVendorCriteria;
+        if (getVendor().isPresent()) {
+            jvmVendorCriteria = getVendor().map(v -> ((DefaultJvmVendorSpec)v).toCriteria()).get();
         } else {
-            jvmVendor = null; // any vendor is acceptable
+            jvmVendorCriteria = null; // any vendor is acceptable
         }
         daemonJvmPropertiesModifier.updateJvmCriteria(
             getPropertiesFile().get().getAsFile(),
-            getJvmVersion().get(),
-            jvmVendor,
-            // TODO fail if empty
+            getLanguageVersion().get(),
+            jvmVendorCriteria,
             getToolchainDownloadUrls().get()
         );
     }
 
+    @SuppressWarnings("Deprecated")
+    private void handleDeprecatedJvmVendor() {
+        if (jvmVendorDeprecated.isPresent()) {
+            String message = "Configuring 'jvmVendor' is no longer supported.";
```

Comment:
@vmadalin This is the String you want to match on and that will remain.

Note that the current output of problems will change in 8.14, so that text may appear in a line with extra text before it.

Reply:
Thinking about this, there's no quick fix that IDE can suggest a side of migrating from `jvmVendor` to `vendor` by modifying the build files. So we are going to treat it as any other deprecation in Gradle, leaving to users to fix it, considering that this is a advanced usage of the Daemon toolchain. 

Instead the one more interesting is the `Toolchain resolvers did not return download URLs`, here IDE can suggest changing the criteria. I'll add this as a todo on my end